File: Java/Problem_071.java
public class Euler71 {
    public static void main(String[] args) {
        final int LIMIT = 1_000_000;
        long bestN = 0, bestD = 1;

        for (long d = 2; d <= LIMIT; d++) {
            long n = (3 * d - 1) / 7; // largest n/d < 3/7
            if (n * bestD > bestN * d) {
                bestN = n;
                bestD = d;
            }
        }

        System.out.println(bestN); // 428570
    }
}

File: Python/Problem_071.py
def euler_71(limit: int = 1_000_000) -> int:
    best_n, best_d = 0, 1

    for d in range(2, limit + 1):
        n = (3 * d - 1) // 7  # largest n/d < 3/7
        if n * best_d > best_n * d:
            best_n, best_d = n, d

    return best_n

if __name__ == "__main__":
    print(euler_71())  # 428570
